osxkeychain/osxkeychain_darwin.go

```diff
@@ -135,30 +135,26 @@ func (h Osxkeychain) List() (map[string]string, error) {
 }
 
 func splitServer(serverURL string) (*C.struct_Server, error) {
-	u, err := url.Parse(serverURL)
+	u, err := parseURL(serverURL)
 	if err != nil {
 		return nil, err
 	}
 
-	hostAndPort := strings.Split(u.Host, ":")
-	host := hostAndPort[0]
+	proto := C.kSecProtocolTypeHTTPS
+	if u.Scheme == "http" {
+		proto = C.kSecProtocolTypeHTTP
+	}
 	var port int
-	if len(hostAndPort) == 2 {
-		p, err := strconv.Atoi(hostAndPort[1])
+	if u.Port() != "" {
```

Comment:
These go1.8 Port() and Hostname() methods are only useful for supporting ipv6, based on what I've read from their code. We could argue it's out of scope for this PR. Or we should provide pre-1.8 functions with `!go1.8` buildtags